tests/test_modeling_tf_common.py

```diff
@@ -461,14 +461,14 @@ def test_compile_tf_model(self):
             if self.is_encoder_decoder:
                 input_ids = {
                     "decoder_input_ids": tf.keras.Input(
-                        batch_shape=(2, 2000), name="decoder_input_ids", dtype="int32"
+                        batch_shape=(2, 512), name="decoder_input_ids", dtype="int32"
                     ),
-                    "input_ids": tf.keras.Input(batch_shape=(2, 2000), name="input_ids", dtype="int32"),
+                    "input_ids": tf.keras.Input(batch_shape=(2, 512), name="input_ids", dtype="int32"),
                 }
             elif model_class in TF_MODEL_FOR_MULTIPLE_CHOICE_MAPPING.values():
-                input_ids = tf.keras.Input(batch_shape=(4, 2, 2000), name="input_ids", dtype="int32")
+                input_ids = tf.keras.Input(batch_shape=(4, 2, 512), name="input_ids", dtype="int32")
             else:
-                input_ids = tf.keras.Input(batch_shape=(2, 2000), name="input_ids", dtype="int32")
+                input_ids = tf.keras.Input(batch_shape=(2, 512), name="input_ids", dtype="int32")
 
```

Comment:
Replace 2000 by 512 here because all the models but Longformer take at most 512 tokens. Also keeping 2000 here will raise an error in the new embedding layers for these models limited to 512 tokens.

Reply:
Maybe we can put it to `self.model_tester.max_position_embeddings` so that it is model specific?

Reply:
Very good idea!! Just done it!

Reply:
@LysandreJik apparently XLNet, TransfoXL and T5 have not a `max_position_embeddings` attribute in their config.

Reply:
Ah, I did not take this into account. Giving them a `max_position_embeddings` wouldn't be wise, though, as they actually do not have such a limitation.

I guess here the best approach would be to get a `max_position_embeddings`, and if they don't have it (they're using relative position embeddings), then set a default of 512 for the sake of this test only.

That means something like `getattr(self.model_tester, "max_position_embeddings", 512)`.

What do you think? This way it stays model-agnostic, but doesn't add the model_tester attribute `max_position_embeddings` as it doesn't belong here.

Reply:
Perfect! I like it 👍

Reply:
Like this?